templates/content/spark.metastore.readme.tt

```diff
@@ -1,53 +1,96 @@
+Use the configuration we generated to point your cluster to lakeFS.
+Now your Spark code can interact with lakeFS:
 
-## Metastore Configuration
+1. Read a dataframe from lakeFS:
 
-The [`lakectl metastore` commands](https://docs.lakefs.io/integrations/glue_hive_metastore.html#commands) can run on Glue or Hive metastore.
-To use `lakectl`, you will need to download the binary. [Learn more](https://www.youtube.com/watch?v=8nO7RT411nA).
+   ```
+   val df = spark.read.parquet("s3a://{{ .Query.repo }}/{{ .Query.branch }}/sensor_data")
+   ```
 
-Add the following to the lakectl configuration file (by default ~/.lakectl.yaml):
+1. Write a dataframe to lakeFS:
 
-### Hive
+   ```
+   df.write.partitionBy("example-column").parquet(s"s3a://{{ .Query.repo }}/{{ .Query.branch }}/output_table/")
+   ```
 
-```conf
-metastore:
-  type: hive
-  hive:
-    uri: <hive-metastore-uri>:<hive-metastore-port>
-```
+### Working with Hive Metastore
+
+In Hive Metastore (or AWS Glue), a table location can only be under a single lakeFS branch.
+For example, consider a table pointing to the `{{ .Query.branch }}` branch:
 
-### Glue
-
-```conf
-metastore:
-  type: glue
-  glue:
-    catalog-id: <catalog id>
-    region: <AWS region>
-    profile: default # optional, implies using a credentials file
-    credentials:
-      access_key_id: <AWS access key id>
-      secret_access_key: <AWS secret access key>
 ```
+CREATE EXTERNAL TABLE `sensor_data` (
+  `ts` timestamp,
+  `value` double
+  PARTITIONED BY (`dt` string)
+  LOCATION 's3a://{{ .Query.repo }}/{{ .Query.branch }}/sensor_data';
+```
+
+Suppose you create a new branch, `new_feature`. You will not be able to access the table from the new branch,
+since your catalog doesn't point to a location in this branch.
+
+To solve this, use the [lakectl metastore commands](https://docs.lakefs.io/integrations/glue_hive_metastore.html#commands):
+
+1. [Download and configure lakectl](https://www.youtube.com/watch?v=8nO7RT411nA).
+
+1. Add the following to the lakectl configuration file (by default ~/.lakectl.yaml`):
+
+   #### Hive
+
+   ```conf
+   metastore:
+     type: hive
+     hive:
+       uri: <hive-metastore-uri>:<hive-metastore-port>
+   ```
+
+   #### Glue
 
-## Suggested model
+   ```conf
+   metastore:
+     type: glue
+     glue:
+       catalog-id: <catalog id>
+       region: <AWS region>
+       profile: default # optional, implies using a credentials file
+       credentials:
+         access_key_id: <AWS access key id>
+         secret_access_key: <AWS secret access key>
+   ```
 
-For simplicity, we recommend creating a schema for each branch. That way, you can use the same table name across different schemas.
-For example, using Trino:
-`CREATE SCHEMA {{ .Query.branch }}_schema with (location = 's3://{{ .Query.repo }}/{{ .Query.branch }}/');`
+1. Use the `lakectl metastore copy` command to create a table for the branch:
```

Comment:
The text is not accurate- it's a schema for the branch.
This will only work if the users already have a `default` schema in their meta store (and obviously a table).